app/src/main/java/com/celzero/bravedns/ui/UniversalFirewallFragment.kt

```diff
@@ -106,22 +106,22 @@ class UniversalFirewallFragment : Fragment(R.layout.universal_fragement_containe
 
         if (DEBUG) Log.d(LOG_TAG, "UniversalFirewallFragment - post observer")
 
-        includeView.firewallAllAppsCheck.isChecked = persistentState.getFirewallModeForScreenState()
+        includeView.firewallAllAppsCheck.isChecked = persistentState.screenState
 
         includeView.firewallUdpConnectionModeCheck.isChecked = persistentState.udpBlockedSettings
         includeView.firewallUnknownConnectionModeCheck.isChecked = persistentState.blockUnknownConnections
 
         includeView.firewallAllAppsCheck.setOnCheckedChangeListener { _, b ->
-            persistentState.setFirewallModeForScreenState(b)
+            persistentState.screenState = b
         }
 
         includeView.firewallAllAppsTxt.setOnClickListener {
-            if (persistentState.getFirewallModeForScreenState()) {
+            if (persistentState.screenState) {
                 includeView.firewallAllAppsCheck.isChecked = false
-                persistentState.setFirewallModeForScreenState(false)
+                persistentState.screenState = false
             } else {
                 includeView.firewallAllAppsCheck.isChecked = true
-                persistentState.setFirewallModeForScreenState(true)
+                persistentState.screenState = true
```

Comment:
The click listener should rely on the check state of the view, not the persistent-state. The persistent-state should reflect whatever the view's checked state is.

```
  persistentState.screenState = view.isChecked
```

Reply:
This is for the check state change when the user clicks on the text view instead of toggle in the Universal firewall screen. 
So based on the click, the value is updated in the UI(toggle) and as well as to the shared preference. 
Modified as below
```
includeView.firewallAllAppsCheck.isChecked = !persistentState.screenState
persistentState.screenState = !persistentState.screenState
```